### PR TITLE
Issues fixed

### LINKED ISSUE #100: Suggest: arrow to navigate between previous/next block.
Suggest to have arrows for easy navigating previous and next block rather than go back to the previous page and click on the next block or edit the URL.

Screen Shot 2019-09-25 at 5 18 05 PM

### PR DESCRIPTION
**Pull request by jomongeorge**
Fixes #100 , #108 ,Style fix

### CODE REVIEW

File: client/src/views/BlockDetail.vue [hunk 1]
```diff
@@ -28,6 +28,20 @@
               <div class="box-title">
                 <h1 class="inline-block">Block</h1>
                 <span class="info_append">{{'# '+this.$route.params.blockId}}</span>
+                <div class="btn_grp inline-block flt-rt">
+                  <ul class="pagination">
+                    <li class="page-item">
+                      <a @click="prevBlcok()" title="Previous Block">
```
Comment (evias) [inline client/src/views/BlockDetail.vue]: fix typo on `prevBlock`

File: client/src/views/BlockDetail.vue [hunk 2]
```diff
@@ -142,48 +153,63 @@ export default {
...
+  },
+  watch: {
+    $route: "asyncData"
   },
   mounted() {
-    this.asyncData()
+    this.asyncData();
   },
   methods: {
+    prevBlcok() {
```
Comment (evias) [inline client/src/views/BlockDetail.vue]: typo

File: client/src/views/BlockDetail.vue (hunk 1 above)
Comment (jomongeorge) [reply client/src/views/BlockDetail.vue]: @evias changed

File: client/src/views/BlockDetail.vue [hunk 3]
```diff
@@ -142,48 +153,63 @@ export default {
...
+  watch: {
+    $route: "asyncData"
   },
   mounted() {
-    this.asyncData()
+    this.asyncData();
   },
   methods: {
+    previousBlcok() {
+      var previousBlcok = Number(this.blockId) - 1;
```
Comment (dgarcia360) [inline client/src/views/BlockDetail.vue]: previousBlcok -> previousBlock

File: client/src/views/BlockDetail.vue [hunk 4]
```diff
@@ -142,48 +153,63 @@ export default {
...
+      this.$router.push({
+        path: "/block/" + previousBlcok
```
Comment (dgarcia360) [inline client/src/views/BlockDetail.vue]: previousBlcok -> previousBlock

File: client/src/views/BlockDetail.vue [hunk 5]
```diff
@@ -28,6 +28,20 @@
               <div class="box-title">
                 <h1 class="inline-block">Block</h1>
                 <span class="info_append">{{'# '+this.$route.params.blockId}}</span>
+                <div class="btn_grp inline-block flt-rt">
+                  <ul class="pagination">
+                    <li class="page-item">
+                      <a @click="previousBlcok()" title="Previous Block">
```
Comment (dgarcia360) [inline client/src/views/BlockDetail.vue]: previousBlcok -> previousBlock

File: client/src/views/BlockDetail.vue [hunk 6]
```diff
@@ -142,48 +153,63 @@ export default {
...
-            temp.push(idx + 1)
-            temp.push(el.deadline)
-            temp.push(el.transactionBody.type)
-            temp.push(el.fee)
-            temp.push(singerLink)
-            temp.push(transactionHashLink)
-            self.tableData.data.push(temp)
-          })
+            let temp = [];
+            let singerLink = `<a href="/#/account/${el.signer}">${el.signer}</a>`;
```
Comment (dgarcia360) [inline client/src/views/BlockDetail.vue]: singerLink --> signerLink

File: client/src/views/BlockDetail.vue [hunk 7]
```diff
@@ -142,48 +153,63 @@ export default {
...
+      });
+    },
     asyncData() {
-      let self = this
-      this.blockId = this.$route.params.blockId
-      this.loading = 0
-      DataService.getBlockInfo(this.blockId).then(function (data) {
-        self.blockInfo = data.blockInfo
+      let self = this;
+      this.blockId = this.$route.params.blockId;
```
Comment (dgarcia360) [inline client/src/views/BlockDetail.vue]: if self = this, why using this.blockId and not self.blockId?

File: client/src/views/BlockDetail.vue [hunk 8]
```diff
@@ -142,48 +153,63 @@ export default {
...
-            temp.push(el.deadline)
-            temp.push(el.transactionBody.type)
-            temp.push(el.fee)
-            temp.push(singerLink)
-            temp.push(transactionHashLink)
-            self.tableData.data.push(temp)
-          })
+            let temp = [];
+            let singerLink = `<a href="/#/account/${el.signer}">${el.signer}</a>`;
+            let transactionHashLink = `<a href="/#/transaction/${el.transactionHash}">${el.transactionHash}</a>`;
```
Comment (dgarcia360) [inline client/src/views/BlockDetail.vue]: This line does not look right, the js code should not be creating hardcoded DOM elements. One option is to pass the necessary parameters to ``self.tableData`` and the HTML reconstructs the URL.

File: client/src/views/BlockDetail.vue [hunk 9]
```diff
@@ -142,48 +153,63 @@ export default {
...
+            temp.push(transactionHashLink);
+            self.tableData.data.push(temp);
+          });
         } else {
-          self.tableData.data = []
+          self.tableData.data = [];
         }
-        self.loading = 1
-      })
+        self.loading = 1;
```
Comment (dgarcia360) [inline client/src/views/BlockDetail.vue]: A previous line inside the same function is defined a "this.loading"

File: client/src/views/BlockDetail.vue [hunk 10]
```diff
@@ -142,48 +153,63 @@ export default {
...
+    },
     asyncData() {
-      let self = this
-      this.blockId = this.$route.params.blockId
-      this.loading = 0
-      DataService.getBlockInfo(this.blockId).then(function (data) {
-        self.blockInfo = data.blockInfo
+      let self = this;
+      this.blockId = this.$route.params.blockId;
+      this.loading = 0;
```
Comment (dgarcia360) [inline client/src/views/BlockDetail.vue]: If loading can only have values 0 or 1, we could use booleans

File: client/src/views/BlockDetail.vue (hunk 3 above)
Comment (jomongeorge) [reply client/src/views/BlockDetail.vue]: ohh sorry, my bad,

File: client/src/views/BlockDetail.vue (hunk 7 above)
Comment (jomongeorge) [reply client/src/views/BlockDetail.vue]: we can't access the vue data inside the function by calling 'this' . Inside the " DataService.getBlockInfo(this.blockId)"  this referred to the function only. 
So we refer **this** to **self** to access inside the function.

File: client/src/views/BlockDetail.vue (hunk 8 above)
Comment (jomongeorge) [reply client/src/views/BlockDetail.vue]: will find a solution to let  the template construct the URL

File: client/src/views/BlockDetail.vue (hunk 7 above)
Comment (dgarcia360) [reply client/src/views/BlockDetail.vue]: I see, thanks!

**Pull request closed without merging**